src/otx/core/model/visual_prompting.py

```diff
@@ -780,162 +813,81 @@ def __init__(
         self.point_labels_box = np.array([[2, 3]], dtype=np.float32)
         self.has_mask_inputs = [np.array([[0.0]]), np.array([[1.0]])]
 
+        self.reference_feats: np.ndarray | None = None
+        self.used_indices: np.ndarray | None = None
+
         self.initialize_reference_info()
 
-    def _create_model(self) -> dict[str, Model]:
+    def _create_model(self) -> SAMLearnableVisualPrompter:
         """Create a OV model with help of Model API."""
-        from model_api.adapters import OpenvinoAdapter, create_core, get_user_config
-        from model_api.models import Model
-
-        ov_models: dict[str, Model] = {}
-
-        plugin_config = get_user_config("AUTO", str(self.num_requests), "AUTO")
+        from model_api.adapters import OpenvinoAdapter, create_core
+
+        ov_device = "CPU"
+        ie = create_core()
+        if not self.force_cpu:
+            devices = ie.available_devices
+            for device in devices:
+                device_name = ie.get_property(device_name=device, property="FULL_DEVICE_NAME")
+                if "dGPU" in device_name and "Intel" in device_name:
+                    ov_device = device
+                    break
+
+        plugin_config = {}
         if self.use_throughput_mode:
             plugin_config["PERFORMANCE_HINT"] = "THROUGHPUT"
-
         model_parameters = {"decoder": {"input_layouts": "image_embeddings:NCHW"}}
+
+        ov_models: dict[str, Model] = {}
         for module in ["image_encoder", "decoder"]:
             model_adapter = OpenvinoAdapter(
                 core=create_core(),
+                device=ov_device,
                 model=self.model_names.get(module),
                 model_parameters=model_parameters.get(module, {}),
                 max_num_requests=self.num_requests,
                 plugin_config=plugin_config,
             )
-            ov_models[module] = Model.create_model(model_adapter, module, configuration=self.model_api_configuration)
-        return ov_models
+            ov_models[module] = Model.create_model(
+                model_adapter,
+                model_type=f"sam_{module}",
+                configuration=self.model_api_configuration,
+            )
+
+        return SAMLearnableVisualPrompter(ov_models["image_encoder"], ov_models["decoder"])
 
     def learn(
         self,
         inputs: ZeroShotVisualPromptingBatchDataEntity,
         reset_feat: bool = False,
-        default_threshold_reference: float = 0.3,
-        is_cascade: bool = False,
     ) -> tuple[dict[str, np.ndarray], list[np.ndarray]]:
         """`Learn` for reference features."""
-        if reset_feat or self.reference_feats is None:
-            self.initialize_reference_info()
+        images, processed_prompts = self._customize_inputs(inputs)
+        reference_masks: list[np.ndarray] = []
+        reference_features = VisualPromptingFeatures(np.array(0), np.array(0))
```

Comment:
Is it needed to initialize `reference_features`?

Reply:
Yes, it's a good practice to store a safe initializer value in a variable which is going to be modified depending on some condition.